langfuse/client.py

```diff
@@ -163,15 +171,15 @@ def __init__(
         )
 
         if not public_key:
-            self.log.warning("public_key is not set.")
-            raise ValueError(
-                "public_key is required, set as a parameter or environment variable 'LANGFUSE_PUBLIC_KEY'"
+            self.enabled = False
+            self.log.warning(
+                "Langfuse client is disabled since no public_key was provided as a parameter or environment variable 'LANGFUSE_PUBLIC_KEY'."
             )
 
         if not secret_key:
-            self.log.warning("secret_key is not set.")
-            raise ValueError(
-                "secret_key is required, set as parameter or environment variable 'LANGFUSE_SECRET_KEY'"
+            self.enabled = False
+            self.log.warning(
```

Comment:
dx improvement: log only a single warning when disabled and state reason in this warning. currently user gets up to 3 warnings